server/__tests__/apollo-server.test.js

```
@@ -37,40 +37,56 @@ describe("Test Cat Wiki API", () => {
   `;
 
   it("should return single breed", async () => {
-    nock(baseURL).get("/breeds").query({ limit: 1 }).reply(200, breed);
+    nock(BASE_URL).get("/breeds").query({ limit: 1 }).reply(200, breeds);
 
     const response = await query({
       query: GET_BREEDS,
       variables: { limit: 1 },
     });
 
     expect(response.errors).toBeUndefined();
-    expect(response).toMatchSnapshot();
+    expect(response.data).toMatchSnapshot();
   });
 
   it("should return breeds by name", async () => {
-    nock(baseURL)
+    nock(BASE_URL)
       .get("/breeds/search")
       .query({ q: "persian" })
-      .reply(200, breed);
+      .reply(200, breeds);
 
     const response = await query({
       query: GET_BREEDS_BY_NAME,
       variables: { breedName: "persian" },
     });
-    expect(response).toMatchSnapshot();
+    expect(response.data).toMatchSnapshot();
   });
 
-  it("should throw error if no breed name is given", async () => {
-    nock(baseURL)
+  it("should return null if breed name does not exist", async () => {
+    nock(BASE_URL)
       .get("/breeds/search")
-      .query({ q: "persian" })
+      .query({ q: "whatever" })
       .reply(200);
 
     const response = await query({
-      query: GET_BREEDS_BY_NAME
+      query: GET_BREEDS_BY_NAME,
+      variables: { breedName: "whatever"}
+    });
+
+    expect(response.data).toBeNull();
+  });
+
+  it("should throw error if no breed name is provided", async () => {
+    nock(BASE_URL)
+      .get("/breeds/search")
+      .query({ q: "" })
+      .reply(200);
```

I don't think that is needed as this should throw a GQL error and not hit the API at all.